### PR TITLE
topgun/k8s worker lifecycle

### PR DESCRIPTION
**Pull request by YoussB**
fixes: #3111 
- Added test for the worker lifecycle in the helm chart.
- this PR is depending on helm/charts#12833

cc @taylorsilva

### CODE REVIEW

File: topgun/k8s/worker_lifecycle_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+			if c.RerunJob {
+				By("running the same job again and watching it to completion")
+				fly.Run("trigger-job", "-j", "some-pipeline/simple-job", "-w")
+			}
+			helmDestroy(releaseName)
+			Wait(Start(nil, "kubectl", "delete", "namespace", namespace, "--wait=false"))
+			Wait(proxySession.Interrupt())
+		},
+		Entry("", Case{
```
Comment (cirocosta) [inline topgun/k8s/worker_lifecycle_test.go]:
```suggestion
		Entry("gracefully", Case{
```

File: topgun/k8s/worker_lifecycle_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+			TerminationGracePeriod:    "600",
+			PipelineYamlFile:          "simple-pipeline.yml",
+			WorkerDisappearingTimeout: 1 * time.Minute,
+			RerunJob:                  true,
+		}),
+		Entry("", Case{
```
Comment (cirocosta) [inline topgun/k8s/worker_lifecycle_test.go]:
```suggestion
		Entry("ungracefully", Case{
```

File: topgun/k8s/worker_lifecycle_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	)
+
+	type Case struct {
+		TerminationGracePeriod    string
+		PipelineYamlFile          string
+		WorkerDisappearingTimeout time.Duration
+		RerunJob                  bool
+	}
+
+	DescribeTable("",
```
Comment (cirocosta) [inline topgun/k8s/worker_lifecycle_test.go]:
```suggestion
	DescribeTable("retiring a worker",
```

**Comment by YoussB** [comment]
@cirocosta turns out that last Friday we were testing against the wrong branch. This PR is good to go now.

**Pull request merged**